### ISSUE TITLE
Cannot map StatsD metrics with consecutive dashes in name

### ISSUE THREAD

**Issue by evandam** [prometheus/statsd_exporter#380]
Trying to reference an incoming StatsD metric with two dashes `--` in the name results in an error starting `statsd-exporter`. Replacing the dashes with an underscore results in the metric not being mapped, though.

If I had to guess, https://github.com/prometheus/statsd_exporter/blob/master/pkg/mapper/mapper.go#L33-L37 should be tweaked to allow `*` dashes.

Example metric `default-ubuntu-1804_kitchen.fabio--requests.count:0|c`

```
level=debug ts=2021-06-16T23:54:45.193Z caller=listener.go:73 msg="Incoming line" proto=udp line=default-ubuntu-1804_kitchen.fabio--requests.count:0|c
```

Mapping with underscores:

```yaml
mappings:
- match: "*.fabio__requests.count"
  name: "fabio_requests_count"
  help: "Fabio request count"
  labels:
    hostname: "$1"
```

Results in autogenerated metric mapping, not what we specify.

```bash
$ curl -s localhost:9102/metrics | grep 'requests_count'
# HELP default_ubuntu_1804_kitchen_fabio___requests_count Metric autogenerated by statsd_exporter.
# TYPE default_ubuntu_1804_kitchen_fabio___requests_count counter
default_ubuntu_1804_kitchen_fabio___requests_count 0
```

Mapping with dashes as seen in incoming metric:

```yaml
mappings:
- match: "*.fabio--requests.count"
  name: "fabio_requests_count"
  help: "Fabio request count"
  labels:
    hostname: "$1"
```

`statsd-exporter` logs:

```
$ statsd_exporter --statsd.mapping-config=mapping.yml --log.level=debug
level=info ts=2021-06-16T23:57:30.868Z caller=main.go:321 msg="Starting StatsD -> Prometheus Exporter" version="(version=0.21.0, branch=HEAD, revision=ef6627b9f05350d54cd3bfea5afe36617d7eb5a4)"
level=info ts=2021-06-16T23:57:30.869Z caller=main.go:322 msg="Build context" context="(go=go1.16.5, user=root@8ace135a0329, date=20210610-07:24:32)"
level=error ts=2021-06-16T23:57:30.870Z caller=main.go:340 msg="error loading config" error="invalid match: *.fabio--requests.count"
```

**Comment by SuperQ** [comment]
Yea, I'm not sure why the dash is outside the character class matching.

**Comment by matthiasr** [comment]
Yes, seems like we can allow this. Will you make a PR @evandam?

**Comment by SuperQ** [comment]
It looks like the multi-dash bad test was added in https://github.com/prometheus/statsd_exporter/pull/24

**Comment by matthiasr** [comment]
Looks like that was relaxing things "a little bit" – if there is a need,
and it doesn't cause downstream issues, I am happy to relax things more.

**Comment by matthiasr** [comment]
This is now fixed for mapped metrics (#381).
